hilti/toolchain/src/compiler/optimizer.cc

```diff
@@ -2155,16 +2156,45 @@ struct FunctionParamVisitor : OptimizerVisitor {
     }
 
     /** Removes the param_id as used within the function. */
-    void removeUsed(type::Function* ftype, const ID& function_id, const ID& param_id) {
+    void removeUsed(type::Function* ftype, const ID& function_id, Expression* name) {
         auto& unused = fn_unused_params.at(function_id);
 
+        std::string_view id = {};
+        const Declaration* resolved_declaration = nullptr;
+
+        if ( auto* x = name->tryAs<expression::Name>() ) {
+            id = x->id().str();
+            assert(x->resolvedDeclaration());
+            resolved_declaration = x->resolvedDeclaration();
+        }
+        else if ( auto* x = name->tryAs<expression::Keyword>() ) {
```

Comment:
could `const` this technically if you want, I wanted to `const` the `name` param but we can't :/


```suggestion
        else if ( const auto* x = name->tryAs<expression::Keyword>() ) {
```